src/ViewLayer/DebugDisplay/BatteryPage/BatteryView/BatteryView.cpp

```diff
@@ -0,0 +1,493 @@
+
+    //connect(&batteryPresenter, SIGNAL(updateProgress(const double, double)), this, SLOT(updateProgress(const double, double)));
+}
+
+
+void BatteryView::aliveReceived(bool alive)
+{
+    if (alive)
+    {
+        ui_.aliveWidget().setStyleSheet(BATTERY_ALIVE);
+    }
+    else
+    {
+        ui_.aliveWidget().setStyleSheet(BATTERY_DEAD);
+    }
+}
+
+void BatteryView::bmsRelayStatusReceived(BmsRelayStatusFlags BmsRelayStatus)
+{
+    if (BmsRelayStatus.alwaysOnSignalStatus())
+    {
+        ui_.BMSOn().setStyleSheet(ON);
+    }
+    else
+    {
+        ui_.BMSOn().setStyleSheet(OFF);
+    }
+
+    if (BmsRelayStatus.isChargingSignalStatus())
+    {
+        ui_.BMSCharging().setStyleSheet(ON);
+    }
+    else
+    {
+        ui_.BMSCharging().setStyleSheet(OFF);
+    }
+
+    if (BmsRelayStatus.isReadySignalStatus())
+    {
+        ui_.BMSReady().setStyleSheet(ON);
+    }
+    else
+    {
+        ui_.BMSReady().setStyleSheet(OFF);
+    }
+
+    if (BmsRelayStatus.dischargeRelayEnabled())
+    {
+        ui_.BMSDischargeRelayEnabled().setStyleSheet(ON);
+    }
+    else
+    {
+        ui_.BMSDischargeRelayEnabled().setStyleSheet(OFF);
+    }
+
+    if (BmsRelayStatus.chargeRelayEnabled())
+    {
+        ui_.BMSChargeRelayEnabled().setStyleSheet(ON);
+    }
+    else
+    {
+        ui_.BMSChargeRelayEnabled().setStyleSheet(OFF);
+    }
+
+    if (BmsRelayStatus.chargerSafetyEnabled())
+    {
+        ui_.BMSChargerSafetyEnabled().setStyleSheet(ON);
+    }
+    else
+    {
+        ui_.BMSChargerSafetyEnabled().setStyleSheet(OFF);
+    }
+
+    if (BmsRelayStatus.malfunctionIndicatorActive())
+    {
+        ui_.malfunctionIndicatorActive().setStyleSheet(ON);
+    }
+    else
+    {
+        ui_.malfunctionIndicatorActive().setStyleSheet(OFF);
+    }
+
+    if (BmsRelayStatus.multiPurposeInputSignalStatus())
+    {
+        ui_.multipurposeInputSignalStatus().setStyleSheet(ON);
+    }
+    else
+    {
+        ui_.multipurposeInputSignalStatus().setStyleSheet(OFF);
+    }
+
+}
+
+void BatteryView::populatedCellsReceived(int populatedCells)
+{
+    ui_.populatedCellsLabel().setNum(populatedCells);
+}
+
+void BatteryView::inputVoltage12VReceived(double inputVoltage)
+{
+    ui_.inputVoltage12VLabel().setText(QString::number(inputVoltage) + " " + VOLTAGE_UNIT);
+}
+
+
+void BatteryView::fanVoltageReceived(double fanVoltage)
+{
+    ui_.fanVoltage().setText(QString::number(fanVoltage) + " " + VOLTAGE_UNIT);
+}
+
+void BatteryView::packCurrentReceived(double packCurrent)
+{
+    ui_.packInfoCurrent().setText(QString::number(packCurrent) + " " + CURRENT_UNIT);
+}
+
+void BatteryView::packVoltageReceived(double packVoltage)
+{
+    ui_.packInfoVoltage().setText(QString::number(packVoltage) + " " + VOLTAGE_UNIT);
+}
+
+void BatteryView::packStateOfChargeReceived(double packStateOfCharge)
+{
+    ui_.progressBarContainer().addWidget(&bar_);
+    bar_.progress = packStateOfCharge / 100;
+    bar_.update();
```

Comment:
Instead of exposing the inner parts of `bar`, let's encapsulate this and give `bar` a single function that the view uses (something like `updateProgress(double value)`)